perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -2593,7 +2594,11 @@ public static synchronized void refreshAuthz(PerunSession sess) {
 			}
 		}
 
-		checkMfaForHavingRole(sess, sess.getPerunPrincipal().getRoles());
+		if (!serviceRole) {
```

Comment:
@xflord @balcirakpeter Please re-think if PerunAdmin should be considered as a service role in the context. 
 I agree with the internal roles like engine, registrar, etc., but do not see why it should be not checked for the PerunAdmin role, given that this role is represented by a person capable of performing Multi-Factor Authentication.

Reply:
(am on holiday and might be out of the loop but will try to help when possible)

Reply:
Ah, ignore my comment, please. I misunderstood the code above. I did not realize that it actually checks the PerunAdmin for service roles, not for every role.
(and of course, I should have put it a few lines above 😁 )